shapeshift/web
# feat: make header dynamic in trade modal

## Description

Makes the header in Trade modal dynamic, so that individual screens can customize it. Below are the Trade screens and their corresponding header titles and buttons (if any):

1. `TradeInput` - "Trade" (no buttons)
2. `TradeConfirm` - "Confirm Details" (with back button)
3. `TradeApproval` - "Trade" (no buttons)
4. `SelectAsset` - "Trade" (with back button)

The approach taken was to drop the `Card` tag (as well as `Card.Header` and `Card.Body`) that surrounded the different Trade screens in `TradeCard`, and have the screens have their own `Card` tag (as well as `Card.Header` and `Card.Body`).

This PR also includes some clean ups encountered while we were at it, such as removing `CardActions`, a duplicate of `TradeCard`.

## Notice

Before submitting a pull request, please make sure you have answered the following:


## Pull Request Type


## Issue (if applicable)

Closes #711.

## Testing

Please outline all testing steps

1. Pull branch locally and run `yarn` to install new deps
2. Run `yarn dev`
3. In Trade modal on right of Dashboard page, check that you can see the TradeInput screen with header "Trade". There should be only one header "Trade", instead of one header "Trade" and one more "Trade" header below.
4. Clicking the token (for both buy or sell) brings you to the SelectAsset screen with header "Trade" and a back button. Clicking the back button brings you back to TradeInput screen.
5. Entering an amount and clicking "Preview Trade" brings you to TradeConfirm screen with header "Confirm Details", instead of "Confirm Trade".
6. TBD Approval screen (locally I basically modified the route navigated to from `TradeConfirm` to `Approval` in TradeInput....)

## Screenshots (if applicable)

Locally I modified the dashboard page to display the old Trade modal (left) and the one in this PR (right). Here are the screenshots:

1. TradeInput: ![TradeInput](https://user-images.githubusercontent.com/96822203/149776013-61022a9f-91a6-4c54-bd8f-15b00554d409.jpg)
2. SelectAsset: ![SelectAsset](https://user-images.githubusercontent.com/96822203/149776011-20fda435-1ea7-4ebd-8847-f33b6242ef51.jpg)
3. Approval: ![Approval](https://user-images.githubusercontent.com/96822203/149776005-e2d4552d-2972-4c48-bf13-ab3f70650c34.jpg)
4. TradeConfirm: ![TradeConfirm](https://user-images.githubusercontent.com/96822203/149776009-19c10968-f7c1-4717-9ef7-f8c810c314b5.jpg)
